packages/blocks/src/api/registration.js

```diff
@@ -25,12 +25,313 @@ import { isValidIcon, normalizeIconObject } from './utils';
+ *                                                 name.
+ * @property {WPBlockTransforms}     [transforms]  Block transformations.
+ */
+
+/**
+ * Editor block alignment options.
+ *
+ * @typedef {('left' | 'center' | 'right' | 'wide' | 'full')} WPBlockAlign
+ */
+
+/**
+ * Editor block attributes object.
+ *
+ * @typedef {Object} WPBlockAttributeOptions
+ *
+ * @property {string}                 attribute Use attribute to extract the
+ *                                              value of an attribute from
+ *                                              markup.
+ * @property {string}                 meta      Attributes may be obtained from
+ *                                              a post’s meta rather than from
+ *                                              the block’s representation in
+ *                                              saved post content. For this, an
+ *                                              attribute is required to specify
+ *                                              its corresponding meta key under
+ *                                              the meta key
+ * @property {string}                 selector  A CSS selector used to parse the
+ *                                              saved block's HTML. The selector
+ *                                              specified can be an HTML tag, or
+ *                                              anything queryable such as a
+ *                                              `class` or `id` attribute.
+ * @property {WPBlockAttributeSource} source    Attribute sources are used to
+ *                                              define how the block attribute
+ *                                              values are extracted from saved
+ *                                              post content. They provide a
+ *                                              mechanism to map from the saved
+ *                                              markup to a JavaScript
+ *                                              representation of a block.
+ * @property {WPBlockAttributeType}   type      The type of the attribute.
+ *
+ * @example
+ * ```js
+ * {
+ *     url: {
+ *         type: 'string',
+ *         source: 'attribute',
+ *         selector: 'img',
+ *         attribute: 'src',
+ *     }
+ * }
+ * // { "url": "https://lorempixel.com/1200/800/" }
+ * ```
+ */
+
+/**
+ * Editor block category options.
+ *
+ * @typedef {Object<string, WPBlockAttributeOptions>} WPBlockAttributes
+ */
+
+/**
+ * Editor block category options.
+ *
+ * @typedef {('text'|'html'|'query'|'meta'|'number'|'string'|'integer')} WPBlockAttributeSource
+ */
+
+/**
+ * Editor block category options.
+ *
+ * @typedef {('null'|'boolean'|'object'|'array'|'number'|'string'|'integer')} WPBlockAttributeType
+ */
+
+/**
+ * Editor block category options.
+ *
+ * @typedef {('common'|'formatting'|'layout'|'widgets'|'embed')} WPBlockCategory
```

Comment:
Since a plugin or theme can add their own custom options, I think we can't assume it would be part of this limited set.

https://developer.wordpress.org/block-editor/developers/filters/block-filters/#managing-block-categories